cloudsync/event.py

```diff
@@ -286,17 +302,15 @@ def _fill_event_path(self, event: Event):
         state = self.state.lookup_oid(self.side, event.oid)
         if state:
             event.path = state[self.side].path
-        if not event.path and event.exists:
-            info = self.provider.info_oid(event.oid)
-            if info:
-                event.path = info.path
-                event.otype = info.otype
-                event.hash = info.hash
+        if not event.path and event.exists is True:
+            self._make_event_accurate(event)
 
     def _notify_on_root_change_event(self, event: Event):
         if self._root_path and self._root_oid:
             if self.provider.root_oid == event.oid:
-                if not event.exists:
+                if not event.accurate:
+                    self._make_event_accurate(event)
```

Comment:
This may slow down onedrive unnecessarily -- onedrive sends an event for each folder all the way up to the root (e.g., adding a file to /a/b/c will result in 4 events, 1 for the new file and 1 for each of the 3 parent folders). I am guessing that is the reason onedrive provider translates drive root events to None (it sees tons of them).

What if we only _make_event_accurate() if event.exists is not True?

Alternatively, could add this extra filtering to the onedrive provider.

Reply:
can we suppress the parent events?

Reply:
looks good.

could probably do some additional filtering in onedrive, but there will be some edge cases (detecting root renames, ensuring there is a state entry for the root folder, etc.)